[reland] Move RPC agents to libtorch

lw opened the pull request:
Summary: Reland of #59939.

Test Plan: CI

Differential Revision: D29193234

mrshenli reviewed (approved):
Chatted with @lw offline, the previous one was reverted due to irrelevant flakes.

lw committed:
[reland] Add some TORCH_API annotations to RPC

Summary: Reland of #59939.

Test Plan: CI

Differential Revision: D29193233

fbshipit-source-id: fdee5578280cdd253d46dacd5f0e351ca4211ceb

lw committed:
[reland] Pass RequestCallback to FaultyPG RPC agent

Summary: Reland of #59939.

Test Plan: CI

Differential Revision: D29193235

fbshipit-source-id: 3723b6dbd3aabcb61736d3dee2f9d2caa886ae87

lw committed:
Fix Windows error in distributed

Differential Revision: D29193014

fbshipit-source-id: 8e04d2aeb0e44c2666616e755d846f11a68f95d2

lw committed:
[reland] Move RPC agents to libtorch (#60170)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/60170

Reland of #59939.

Test Plan: CI

Reviewed By: mrshenli

Differential Revision: D29193234

fbshipit-source-id: 689b3ede8945a7f191c8c2b8a7d4ac51add63c44
